src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -151,21 +162,25 @@ internal BindingDataResolver GetAutoResolveResolver(string originalValue, AutoRe
             }
         }
 
-        // Apply AppSetting attribute. 
-        internal static BindingDataResolver GetAppSettingResolver(string originalValue, AppSettingAttribute attr, INameResolver nameResolver, PropertyInfo propInfo, Validator validator)
+        // Both ConnectionString and AppSetting have the same behavior, but perform the lookup differently.
+        internal static BindingDataResolver GetConfigurationResolver(string propertyValue, string defaultValue, PropertyInfo propInfo, Validator validator, Func<string, string> resolveValue)
         {
-            string appSettingName = originalValue ?? attr.Default;
-            string resolvedValue = string.IsNullOrEmpty(appSettingName) ?
-                originalValue : nameResolver.Resolve(appSettingName);
+            string connectionStringName = propertyValue ?? defaultValue;
```

Comment:
nit: this should probably be named something like `configurationKey` or `configurationName` for clarity, as it won't always be a connection string.

Reply:
Done.